osu.Desktop/DiscordRichPresence.cs

```diff
@@ -30,6 +32,9 @@ internal class DiscordRichPresence : Component
 
         private IBindable<APIUser> user;
 
+        [Resolved]
+        private IAPIProvider api { get; set; }
```

Comment:
Also just noticed that this is resolved both to a field here and in to a parameter in `load()` (named `provider`) therein. Would be preferred to rename this field to `provider` and remove the `load()` parameter to unify usage.